### ISSUE TITLE
Update Documentation

### ISSUE THREAD

**Issue by phaabe** [Graylog2/graylog-docker#136]
# Documentation of Graylog v4 contains examples with v3
https://docs.graylog.org/en/4.0/pages/installation/docker.html

## Expected Behavior
Show examples containing graylog docker containers of version 4 (e.g. 4.0.1-1)

## Current
https://docs.graylog.org/en/4.0/pages/installation/docker.html
Examples provide graylog version 3.3

## Possible Solution 1
https://docs.graylog.org/en/4.0/pages/installation/docker.html
Update Documentation with a working examples and correct dependencies for Version 4. 

## Possible Solution 2
Insert a [link to description of version](https://docs.graylog.org/en/4.0/pages/installation.html) dependencies.

**Comment by atc0005** [comment]
It's a tangent, but as part of the 4.0.0 to 4.0.1 Graylog upgrade I opted to upgrade the MongoDB container used with Graylog from 3.6 as the documentation suggested to 4.2. 

I made the mistake of assuming that the Docker container would take care of handling any 3.6 to 4.2 upgrade concerns, but after checking the error logs I saw that I had to go from 3.6 to 4.0 first, *then* I could upgrade from 4.0 to 4.2, but only after manual intervention (AFAICT) to make the upgrade as safe.

https://docs.mongodb.com/manual/release-notes/4.2-upgrade-standalone/

I don't know enough about MongoDB to feel comfortable upgrading 4.0. to 4.2, so any tips for that in the official docs (or just a warning not to attempt a direct 3.6 to 4.2 upgrade) would be helpful to future readers.
